contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -60,9 +75,22 @@ def is_non_synthetic_python_target(target):
   def is_python_target(target):
     return isinstance(target, PythonTarget)
 
+  def _is_tagged_target(self, target):
+    return self.get_options().whitelisted_tag_name in target.tags
+
+  def _is_tagged_non_synthetic_python_target(self, target):
+    return (self.is_non_synthetic_python_target(target) and
+            self._is_tagged_target(target))
+
+  def _all_targets_in_context_are_whitelisted(self, targets, targets_in_context):
+    return len(targets) == 0 or len(targets) == len(targets_in_context)
+
   def _calculate_python_sources(self, targets):
-    """Generate a set of source files from the given targets."""
-    python_eval_targets = filter(self.is_non_synthetic_python_target, targets)
+    """Filter targets to generate a set of source files from the given targets."""
+    whitelisted_targets = Target.closure_for_targets(list(filter(self._is_tagged_target, targets)))
+    python_eval_targets = list(filter(self._is_tagged_non_synthetic_python_target, list(whitelisted_targets)))
+    if not self._all_targets_in_context_are_whitelisted(python_eval_targets, whitelisted_targets):
+      raise MypyTaskError("All targets in context need to be whitelisted for mypy to run")
```

Comment:
As decided in the meeting, this should not be the case. At most, log a warning. We don't want this because it makes it far harder for upstream users of some `python_library` to be able to use type checks, until their downstream dependency has fully type checked. In general, I think we will want to stick with MyPy's default of untyped code using type `Any` and not logging anything extra about it. We could choose to add those warnings if there's a strong push for it, but shouldn't do it otherwise.